adapters/tlsio_ssl_esp8266.c

```diff
@@ -0,0 +1,960 @@
+{
+    int result = 0;
+    unsigned char buffer[64];
+
+    int rcv_bytes = 1;
+    rcv_bytes = SSL_read(tls_io_instance->ssl, buffer, sizeof(buffer));
+    os_printf("ssl recv bytes: %d\n", rcv_bytes);
+    if (rcv_bytes > 0)
+    {
+        if (tls_io_instance->on_bytes_received == NULL)
+        {
+            LogError("NULL on_bytes_received.");
+        }
+        else
+        {
+            tls_io_instance->on_bytes_received(tls_io_instance->on_bytes_received_context, buffer, rcv_bytes);
+        }
+    }
+
+    return result;
+}
+
+static void destroy_openssl_instance(TLS_IO_INSTANCE* tls_io_instance)
+{
+    if (tls_io_instance != NULL)
+    {
+        SSL_free(tls_io_instance->ssl);
+        SSL_CTX_free(tls_io_instance->ssl_context);
+    }
+}
+
+static int add_certificate_to_store(TLS_IO_INSTANCE* tls_io_instance, const char* certValue)
+{
+    return 0;
+}
+
+
+static int create_openssl_instance(TLS_IO_INSTANCE* tlsInstance)
+{
+    return 0;
+}
+
+int tlsio_openssl_init(void)
+{
+    return 0;
+}
+
+void tlsio_openssl_deinit(void)
+{
+    openssl_dynamic_locks_uninstall();
+    openssl_static_locks_uninstall();
+
+    ERR_free_strings();
+}
+
+CONCRETE_IO_HANDLE tlsio_openssl_create(void* io_create_parameters)
+{
+    TLSIO_CONFIG* tls_io_config = io_create_parameters;
+    TLS_IO_INSTANCE* result;
+
+    if (tls_io_config == NULL)
+    {
+        result = NULL;
+        LogError("NULL tls_io_config.");
+    }
+    else
+    {
+        result = malloc(sizeof(TLS_IO_INSTANCE));
+        if (result == NULL)
+        {
+            LogError("Failed allocating TLSIO instance.");
+        }
+        else
+        {
+            memset(result, 0, sizeof(TLS_IO_INSTANCE));
+            mallocAndStrcpy_s(&result->hostname, tls_io_config->hostname);
+            result->port = tls_io_config->port;
+            result->ssl_context = NULL;
+            result->ssl = NULL;
+            result->certificate = NULL;
+
+            result->on_bytes_received = NULL;
+            result->on_bytes_received_context = NULL;
+
+            result->on_io_open_complete = NULL;
+            result->on_io_open_complete_context = NULL;
+
+            result->on_io_close_complete = NULL;
+            result->on_io_close_complete_context = NULL;
+
+            result->on_io_error = NULL;
+            result->on_io_error_context = NULL;
+
+            result->tlsio_state = TLSIO_STATE_NOT_OPEN;
+
+            result->x509certificate = NULL;
+            result->x509privatekey = NULL;
+        }
+    }
+
+    return result;
+}
+
+void tlsio_openssl_destroy(CONCRETE_IO_HANDLE tls_io)
+{
+    if (tls_io == NULL)
+    {
+        LogError("NULL tls_io.");
+    }
+    else
+    {
+        TLS_IO_INSTANCE* tls_io_instance = (TLS_IO_INSTANCE*)tls_io;
+        if (tls_io_instance->certificate != NULL)
+        {
+            free(tls_io_instance->certificate);
+            tls_io_instance->certificate = NULL;
```

Comment:
This is the same as [tlsio_openssl.c](https://github.com/Azure/azure-c-shared-utility/blob/develop/src/tlsio_openssl.c#L1010). I think it's always a good idea to initialize variables. I have experienced many bugs because of uninitialized variables.